### PR TITLE
Adds Tailoring to the Barber job. Reworks Barber haircuts. New uniform for the Barber to replace codersprites.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request
ShareX_JAwcBVZtbm
Barber haircuts now function via an intuitive tgui interface along with a preview!
Barbers have been given a new uniform, sprited by Fury McFlurry.
Barbers now have a Sewing Machine and Pattern Table in their workspaces, allowing them to produce and design custom clothing! Custom clothing functions via .dmi files.
dreammaker_aqsAlkxF66
Set up your DMI file as shown in the image provided. Note that onmob_digit is optional, but if included, allows your clothing to have digitigrade support.
A video of the in-game process is available at 
https://file.house/VA17.mp4
Things that will block your .dmi:

- Uploading something that isn't a .dmi
- Uploading a broken .dmi
- Uploading a .dmi that isn't 32x32
- Uploading a .dmi that's too large.
- Uploading a .dmi that doesn't have an onmob or inventory state.
- Uploading a .dmi that has more than 3 icon states.

Mannequins are provided in Barber shops to wear clothing.

## How This Contributes To The Skyrat Roleplay Experience

Haircuts are now much easier to do and much easier for everyone involved.
Custom clothing allows artists to create cool clothing for players to use in a way that allows them to go get the clothing in game.
Administrative tools are available to ban individual patterns and ban users from uploading patterns.

## Changelog

:cl: Iamgoofball, Fury McFlurry, Azlan
add: Adds Tailoring to the Barber job! Sprites for the machinery by Fury McFlurry!
add: Custom Clothing can now be produced by the Barber via the Pattern Table and its database, and the Sewing Machine. Sprites by Fury McFlurry.
imageadd: Resprites the Barber's uniform to replace the previous codersprites, and resprites the Scissors and Barber Pole. Sprites by Azlan.
code: Refactors the internals of the barber scissors.
refactor: Barber haircuts now function via a tgui interface.
admin: Adds administrative tools to ban players from uploading clothing, and ban individual patterns.
/:cl:

### CODE REVIEW

**Comment by Gandalf2k15** [comment]
FAIL: /datum/unit_test/subsystem_init 0s
	REASON #1: Clothing Database(/datum/controller/subsystem/clothing_database) is a subsystem meant to initialize but doesn't get set as initialized.

**Comment by Iamgoofball** [comment]
fixed lol

**Comment by GoldenAlpharex** [comment]
Slapping those on until the Headmins can discuss it and come to a conclusion, as it has repercussions on the staff workload, amongst other things.

**Comment by MaxStirlitz** [comment]
That's impressive work! 
But there is a few issues:

In metastation massage parlour, lower switch is controling wrong curtains
image


When you use scissors in your hand, pops up NtOS screen
image

Sewed clothes can not be thrown or be placed on tables

Scisors cannot be hold in medkit(unlike other barber tools)

**Comment by BurgerLUA** [comment]
There are a bit of interaction (clicking) issues with tailored clothing. They can't be placed in bags or on tables. There is probably more but I don't want to risk getting stuck.

**Comment by Iamgoofball** [comment]
Sorry guys, I fixed a bunch of this but the testmerge never got updated, next round :tm: it'll be fixed

**Comment by BurgerLUA** [comment]
> Sorry guys, I fixed a bunch of this but the testmerge never got updated, next round ™️ it'll be fixed

Nice.

Also another bug, jumpsuits don't currently display the proper worn sprite. I think that might've been fixed, though.

File: modular_skyrat/modules/salon/code/pattern_table.dm [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	return
+
+/obj/machinery/pattern_table/proc/create_pattern(mob/living/user, list/modifiers)
+	var/has_dmi = tgui_alert(user, "Do you have a prepared .dmi file?", "Pattern Design", list("Yes", "No"))
+	if(!has_dmi || has_dmi == "No")
+		user.balloon_alert(user, "see chat!")
+		to_chat(user, "Please see [DMI_TUTORIAL_URL] for an example of a properly set up 32x32 .dmi file from inside Dream Maker.")
+		to_chat(user, "If the URL does not function for you, your .dmi file will need an onmob icon state with the 4 cardinal directions, and an inventory icon state.")
+		to_chat(user, "If you want your clothing to support digitigrade, you can add an onmob_digit icon state. This is optional, and isn't required.")
+		to_chat(user, "Clothing .dmi files must be 32x32.")
```
Comment (OrionTheFox) [inline modular_skyrat/modules/salon/code/pattern_table.dm:109]:
```suggestion
		to_chat(user, span_warning("Please see [DMI_TUTORIAL_URL] for an example of a properly set up 32x32 .dmi file from inside Dream Maker."))
		to_chat(user, span_warning("If the URL does not function for you, your .dmi file will need an 'onmob' icon state with the 4 cardinal directions, and an 'inventory' icon state."))
		to_chat(user, span_warning("If you want your clothing to support digitigrade, you can add an 'onmob_digit' icon state. This is optional, and isn't required."))
		to_chat(user, span_warning("Clothing .dmi files must be 32x32."))
```
Catches the eye more with the warning(and doesnt appear as plain text), and the 'quotes' make it easier to tell thats the icon state's NAME and not just the description

File: modular_skyrat/modules/salon/code/tailor_clothing_subsystem.dm [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/datum/tailor_clothing
+	var/name = "Clothing Name"
+	var/desc = "Clothing Description"
+	/// The ckey that uploaded this.
+	var/ckey_author
```
Comment (Pirill) [inline modular_skyrat/modules/salon/code/tailor_clothing_subsystem.dm:7]: Gonna second this. It'd help avoid a ton of bloat from minor edits or revisions.

Comment (Iamgoofball) [reply modular_skyrat/modules/salon/code/tailor_clothing_subsystem.dm:7]: Working on this for the next UI update, yeah.

**Comment by TetrakoHateThis** [comment]
So, uh... how we doing with this?

**Comment by Iamgoofball** [comment]
yeah sorry i just need to get this updated

**Comment by TetrakoHateThis** [comment]
Maybe one day.

**Comment by YakumoChen** [comment]
Travesty
Goof please

**Pull request closed without merging**